tests/telemetry/test_writer.py

```diff
@@ -813,3 +813,64 @@ def test_telemetry_writer_is_using_agent_by_default_if_api_key_is_not_available(
     assert telemetry_writer._client._telemetry_url in ("http://localhost:9126", "http://testagent:9126")
     assert "dd-api-key" not in telemetry_writer._client._headers
+
+
+def test_otel_config_telemetry(test_agent_session, run_python_code_in_subprocess, tmpdir):
+    """
+    asserts that telemetry data is submitted for OpenTelemetry configurations
+    """
+
+    env = os.environ.copy()
+    env["DD_SERVICE"] = "dd_service"
+    env["OTEL_SERVICE_NAME"] = "otel_service"
+    env["OTEL_LOG_LEVEL"] = "DEBUG"
+    env["OTEL_PROPAGATORS"] = "tracecontext"
+    env["OTEL_TRACES_SAMPLER"] = "always_on"
+    env["OTEL_TRACES_EXPORTER"] = "none"
+    env["OTEL_LOGS_EXPORTER"] = "otlp"
+    env["OTEL_RESOURCE_ATTRIBUTES"] = "team=apm,component=web"
+    env["OTEL_SDK_DISABLED"] = "true"
+    env["OTEL_UNSUPPORTED_CONFIG"] = "value"
+    env["_DD_INSTRUMENTATION_TELEMETRY_TESTS_FORCE_APP_STARTED"] = "true"
+
+    _, stderr, status, _ = run_python_code_in_subprocess("import ddtrace", env=env)
+    assert status == 0, stderr
+
+    configurations = {c["name"]: c for c in test_agent_session.get_configurations()}
+
+    assert configurations["DD_SERVICE"] == {"name": "DD_SERVICE", "origin": "env_var", "value": "dd_service"}
+    assert configurations["OTEL_LOG_LEVEL"] == {"name": "OTEL_LOG_LEVEL", "origin": "env_var", "value": "debug"}
+    assert configurations["OTEL_PROPAGATORS"] == {
+        "name": "OTEL_PROPAGATORS",
+        "origin": "env_var",
+        "value": "tracecontext",
+    }
+    assert configurations["OTEL_TRACES_SAMPLER"] == {
+        "name": "OTEL_TRACES_SAMPLER",
+        "origin": "env_var",
+        "value": "always_on",
+    }
+    assert configurations["OTEL_TRACES_EXPORTER"] == {
+        "name": "OTEL_TRACES_EXPORTER",
+        "origin": "env_var",
+        "value": "none",
+    }
+    assert configurations["OTEL_LOGS_EXPORTER"] == {"name": "OTEL_LOGS_EXPORTER", "origin": "env_var", "value": "otlp"}
+    assert configurations["OTEL_RESOURCE_ATTRIBUTES"] == {
+        "name": "OTEL_RESOURCE_ATTRIBUTES",
+        "origin": "env_var",
+        "value": "team=apm,component=web",
+    }
+    assert configurations["OTEL_SDK_DISABLED"] == {"name": "OTEL_SDK_DISABLED", "origin": "env_var", "value": "true"}
+
+    env_hiding_metrics = test_agent_session.get_metrics("otel.env.hiding")
+    tags = [m["tags"] for m in env_hiding_metrics]
+    assert tags == [["config_opentelemetry:otel_service_name", "config_datadog:dd_service"]]
+
+    env_unsupported_metrics = test_agent_session.get_metrics("otel.env.unsupported")
+    tags = [m["tags"] for m in env_unsupported_metrics]
+    assert tags == [["config_opentelemetry:otel_unsupported_config"]]
+
+    env_invalid_metrics = test_agent_session.get_metrics("otel.env.invalid")
+    tags = [m["tags"] for m in env_invalid_metrics]
+    assert tags == [["config_opentelemetry:otel_logs_exporter", "config_datadog:"]]
```

Comment:
I don't think the `config_datadog` tag is supposed to be present, can you verify this?